File: source/aoBlogs2/Models/Db/VisitModel.vb


Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses

Namespace Models
    Public Class VisitModel        '<------ set set model Name and everywhere that matches this string
        Inherits DbModel
        Implements ICloneable
        '
        '====================================================================================================
        '-- const
        Public Const contentName As String = "Visits"      '<------ set content name
        Public Const contentTableName As String = "ccVisits"   '<------ set to tablename for the primary content (used for cache names)
        Private Shadows Const contentDataSource As String = "default"             '<------ set to datasource if not default
        '
        '====================================================================================================
        ' -- instance properties
        'instancePropertiesGoHere
        Public Property Bot As Boolean
        'Public Property Browser As String
        'Public Property CookieSupport As Boolean
        Public Property ExcludeFromAnalytics As Boolean
        'Public Property HTTP_REFERER As String
        'Public Property LastVisitTime As Date
        'Public Property LoginAttempts As Integer
        'Public Property MemberID As Integer
        'Public Property MemberNew As Boolean
        'Public Property Mobile As Boolean
        'Public Property PageVisits As Integer
        'Public Property RefererPathPage As String
        'Public Property REMOTE_ADDR As String
        'Public Property StartDateValue As Integer
        'Public Property StartTime As Date
        'Public Property StopTime As Date
        'Public Property TimeToLastHit As Integer
        'Public Property VerboseReporting As Boolean
        'Public Property VisitAuthenticated As Boolean
        'Public Property VisitorID As Integer
        'Public Property VisitorNew As Boolean
        '
        '====================================================================================================
        Public Overloads Shared Function add(cp As CPBaseClass) As VisitModel
            Return add(Of VisitModel)(cp)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cp As CPBaseClass, recordId As Integer) As VisitModel
            Return create(Of VisitModel)(cp, recordId)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cp As CPBaseClass, recordGuid As String) As VisitModel
            Return create(Of VisitModel)(cp, recordGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createByName(cp As CPBaseClass, recordName As String) As VisitModel
            Return createByName(Of VisitModel)(cp, recordName)
        End Function
        '
        '====================================================================================================
        Public Overloads Sub save(cp As CPBaseClass)
            MyBase.save(Of VisitModel)(cp)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Sub delete(cp As CPBaseClass, recordId As Integer)
            delete(Of VisitModel)(cp, recordId)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Sub delete(cp As CPBaseClass, ccGuid As String)
            delete(Of VisitModel)(cp, ccGuid)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cp As CPBaseClass, sqlCriteria As String, Optional sqlOrderBy As String = "id") As List(Of VisitModel)
            Return createList(Of VisitModel)(cp, sqlCriteria, sqlOrderBy)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordName(cp As CPBaseClass, recordId As Integer) As String
            Return DbModel.getRecordName(Of VisitModel)(cp, recordId)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordName(cp As CPBaseClass, ccGuid As String) As String
            Return DbModel.getRecordName(Of VisitModel)(cp, ccGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordId(cp As CPBaseClass, ccGuid As String) As Integer
            Return DbModel.getRecordId(Of VisitModel)(cp, ccGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getCount(cp As CPBaseClass, sqlCriteria As String) As Integer
            Return DbModel.getCount(Of VisitModel)(cp, sqlCriteria)
        End Function
        '
        '====================================================================================================
        Public Overloads Function getUploadPath(fieldName As String) As String
            Return MyBase.getUploadPath(Of VisitModel)(fieldName)
        End Function
        '
        '====================================================================================================
        '
        Public Function Clone(cp As CPBaseClass) As VisitModel
            Dim result As VisitModel = DirectCast(Me.Clone(), VisitModel)
            result.id = cp.Content.AddRecord(contentName)
            result.ccguid = cp.Utils.CreateGuid()
            result.save(of BlogModel)(cp)
            Return result
        End Function
        '
        '====================================================================================================
        '
        Public Function Clone() As Object Implements ICloneable.Clone
            Return Me.MemberwiseClone()
        End Function
        '
        '====================================================================================================
        ''' <summary>
        ''' Save a list of this model to the database, guid required, using the guid as a key for update/import, and ignoring the id.
        ''' </summary>
        ''' <param name="cp"></param>
        ''' <param name="modelList">A dictionary with guid as key, and this model as object</param>
        Public Shared Sub migrationImport(cp As CPBaseClass, modelList As Dictionary(Of String, VisitModel))
            Dim ContentControlID As Integer = cp.Content.GetID(contentName)
            For Each kvp In modelList
                If (Not String.IsNullOrEmpty(kvp.Value.ccguid)) Then
                    kvp.Value.id = 0
                    Dim dbData As VisitModel = create(cp, kvp.Value.ccguid)
                    If (dbData IsNot Nothing) Then
                        kvp.Value.id = dbData.id
                    Else
                        kvp.Value.DateAdded = Now
                        kvp.Value.CreatedBy = 0
                    End If
                    kvp.Value.ContentControlID = ContentControlID
                    kvp.Value.ModifiedDate = Now
                    kvp.Value.ModifiedBy = 0
                    kvp.Value.save(of BlogModel)(cp)
                End If
            Next
        End Sub


    End Class
End Namespace

File: source/aoBlogs2/Models/Db/VisitModel.cs

//using System;
//using System.Collections.Generic;
//using Contensive.BaseClasses;
//using Contensive.Models.Db;

//namespace Contensive.Addons.Blog.Models {
//    public class VisitModel : Contensive.Models.Db.DbBaseModel, ICloneable {        // <------ set set model Name and everywhere that matches this string
//        // 
//        // ====================================================================================================
//        public static DbBaseTableMetadataModel tableMetadata { get; private set; } = new DbBaseTableMetadataModel("xxxxx", "xxxxx", "default", false);
//        // -- const
//        public const string contentName = "Visits";      // <------ set content name
//        public const string contentTableName = "ccVisits";   // <------ set to tablename for the primary content (used for cache names)
//        private  const string contentDataSource = "default";             // <------ set to datasource if not default
//        // 
//        // ====================================================================================================
//        // -- instance properties
//        // instancePropertiesGoHere
//        public bool Bot { get; set; }
//        // Public Property Browser As String
//        // Public Property CookieSupport As Boolean
//        public bool ExcludeFromAnalytics { get; set; }
//        // Public Property HTTP_REFERER As String
//        // Public Property LastVisitTime As Date
//        // Public Property LoginAttempts As Integer
//        // Public Property MemberID As Integer
//        // Public Property MemberNew As Boolean
//        // Public Property Mobile As Boolean
//        // Public Property PageVisits As Integer
//        // Public Property RefererPathPage As String
//        // Public Property REMOTE_ADDR As String
//        // Public Property StartDateValue As Integer
//        // Public Property StartTime As Date
//        // Public Property StopTime As Date
//        // Public Property TimeToLastHit As Integer
//        // Public Property VerboseReporting As Boolean
//        // Public Property VisitAuthenticated As Boolean
//        // Public Property VisitorID As Integer
//        // Public Property VisitorNew As Boolean
//        // 
//        // ====================================================================================================
//        public static VisitModel @add(CPBaseClass cp) {
//            return DbBaseModel.addDefault<VisitModel>(cp);
//        }
//        // 
//        // ====================================================================================================
//        public static VisitModel create(CPBaseClass cp, int recordId) {
//            return create<VisitModel>(cp, recordId);
//        }
//        // 
//        // ====================================================================================================
//        public static VisitModel create(CPBaseClass cp, string recordGuid) {
//            return create<VisitModel>(cp, recordGuid);
//        }
//        // 
//        // ====================================================================================================
//        public static VisitModel createByName(CPBaseClass cp, string recordName) {
//            return createByName<VisitModel>(cp, recordName);
//        }
//        // 
//        // ====================================================================================================
//        public void save(CPBaseClass cp) {
//            save<VisitModel>(cp);
//        }
//        // 
//        // ====================================================================================================
//        public static void delete(CPBaseClass cp, int recordId) {
//            delete<VisitModel>(cp, recordId);
//        }
//        // 
//        // ====================================================================================================
//        public static void delete(CPBaseClass cp, string ccGuid) {
//            delete<VisitModel>(cp, ccGuid);
//        }
//        // 
//        // ====================================================================================================
//        public static List<VisitModel> createList(CPBaseClass cp, string sqlCriteria, string sqlOrderBy = "id") {
//            return createList<VisitModel>(cp, sqlCriteria, sqlOrderBy);
//        }
//        // 
//        // ====================================================================================================
//        public static string getRecordName(CPBaseClass cp, int recordId) {
//            return getRecordName<VisitModel>(cp, recordId);
//        }
//        // 
//        // ====================================================================================================
//        public static string getRecordName(CPBaseClass cp, string ccGuid) {
//            return getRecordName<VisitModel>(cp, ccGuid);
//        }
//        // 
//        // ====================================================================================================
//        public static int getRecordId(CPBaseClass cp, string ccGuid) {
//            return getRecordId<VisitModel>(cp, ccGuid);
//        }
//        // 
//        // ====================================================================================================
//        public static int getCount(CPBaseClass cp, string sqlCriteria) {
//            return getCount<VisitModel>(cp, sqlCriteria);
//        }
//        // 
//        // ====================================================================================================
//        public string getUploadPath(string fieldName) {
//            return getUploadPath<VisitModel>(fieldName);
//        }
//        // 
//        // ====================================================================================================
//        // 
//        public VisitModel Clone(CPBaseClass cp) {
//            VisitModel result = (VisitModel)Clone();
//            result.id = cp.Content.AddRecord(contentName);
//            result.ccguid = cp.Utils.CreateGuid();
//            result.save(cp);
//            return result;
//        }
//        // 
//        // ====================================================================================================
//        // 
//        public object Clone() {
//            return MemberwiseClone();
//        }
//        // 
//        // ====================================================================================================
//        /// <summary>
//        /// Save a list of this model to the database, guid required, using the guid as a key for update/import, and ignoring the id.
//        /// </summary>
//        /// <param name="cp"></param>
//        /// <param name="modelList">A dictionary with guid as key, and this model as object</param>
//        public static void migrationImport(CPBaseClass cp, Dictionary<string, VisitModel> modelList) {
//            int ContentControlID = cp.Content.GetID(contentName);
//            foreach (var kvp in modelList) {
//                if (!string.IsNullOrEmpty(kvp.Value.ccguid)) {
//                    kvp.Value.id = 0;
//                    var dbData = create(cp, kvp.Value.ccguid);
//                    if (dbData is not null) {
//                        kvp.Value.id = dbData.id;
//                    }
//                    else {
//                        kvp.Value.dateAdded = DateTime.Now;
//                        kvp.Value.createdBy = 0;
//                    }
//                    kvp.Value.contentControlId = ContentControlID;
//                    kvp.Value.modifiedDate = DateTime.Now;
//                    kvp.Value.modifiedBy = 0;
//                    kvp.Value.save(cp);
//                }
//            }
//        }


//    }
//}